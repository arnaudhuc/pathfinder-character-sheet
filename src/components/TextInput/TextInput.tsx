/** @jsx jsx */
import { jsx } from '@emotion/core';
import { TextField } from '@material-ui/core';

interface ITextInput {
	label?: string;
}

export const TextInput: React.FC<ITextInput> = (props) => {
	const { label = '' } = props;

	return (
		<div css={{ display: 'flex' }}>
			<TextField label={label} />
		</div>
	);
};
