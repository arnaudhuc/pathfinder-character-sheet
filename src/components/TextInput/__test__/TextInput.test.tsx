/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer';
import { TextInput } from '../TextInput';

describe('TextInput', () => {
	test('should render with default props', () => {
		const container = renderer.create(<TextInput label="Nom" />);

		expect(container).toMatchSnapshot();
	});

	test('should render with no label', () => {
		const container = renderer.create(<TextInput />);

		expect(container).toMatchSnapshot();
	});
});
