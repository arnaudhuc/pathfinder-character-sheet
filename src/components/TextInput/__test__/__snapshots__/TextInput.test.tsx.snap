// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render with default props 1`] = `
<div
  className="css-k008qs"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      Nom
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`TextInput should render with no label 1`] = `
<div
  className="css-k008qs"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;
